lib/chai/utils/getOperator.js

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Module dependencies
+ */
+
+var inspect = require('./inspect');
+var config = require('../config');
+var flag = require('./flag');
+
+function isObjectType(obj) {
+  var str = inspect(obj),
+    type = Object.prototype.toString.call(obj);
+
+  if (config.truncateThreshold && str.length >= config.truncateThreshold) {
```

Comment:
I might be missing something, but why does the truncate threshold influence whether we consider something an `objectType` or not?

Given that the `truncateThreshold` is here to avoid having diffs/messages that are too long and not for dealing with the `Error` thrown, perhaps we should simply have something like:

```js
// The line below was written considering `type-detect` being used
var objectTypes = ['Array', 'Object', 'function'];
return objectTypes.indexOf(type) !== -1;
```